gemma-teixido/challenge/tour-of-heroes-fullstack/backend/src/controllers/heroesController.js

```diff
@@ -0,0 +1,72 @@
+// const nextHeroId = 31;
+const chalk = require('chalk');
+const debug = require('debug')('server:heroesConntroller');
+const Hero = require('../../model/heroModel');
+
+function heroesController() {
+  async function getAll(req, res) {
+    debug(`Dentro de la funcion ${chalk.yellowBright('getAll')}`);
+    const heroes = await Hero.find({});
+    res.json(heroes);
+  }
+
+  async function createOne(req, res) {
+    const newHero = new Hero(req.body);
+    debug(newHero);
+    try {
+      await newHero.save();
+      res.json(newHero);
+    } catch (error) {
+      debug(error);
+      res.send(error);
+    }
+  }
+
+  async function getById(req, res) {
+    try {
+      const heroById = await Hero.findById(
+        req.params.heroId
+      );
+      res.json(heroById);
+    } catch (error) {
+      debug(error);
+      res.status(404);
+      res.send(error);
+    }
+  }
+
+  async function updateById(req, res) {
+    try {
+      const updatedHero = await Hero.findByIdAndUpdate(
+        req.params.heroId,
+        req.body,
+        { new: true }
+      );
+      res.json(updatedHero);
+    } catch (error) {
+      debug(error);
```

Comment:
Try to give back an status code, otherwise the front cannot handle the errors as it does not find the status code of any error! :)